compiler/frontend/beam/src/main/java/org/apache/nemo/compiler/frontend/beam/NemoPipelineResult.java

```diff
@@ -49,10 +60,12 @@ public State cancel() throws IOException {
 
   @Override
   public State waitUntilFinish(final Duration duration) {
-    throw new UnsupportedOperationException();
-    // TODO #208: NemoPipelineResult#waitUntilFinish hangs
-    // Previous code that hangs the job:
-    // return (State) super.waitUntilJobFinish(duration.getMillis(), TimeUnit.MILLISECONDS);
+    try {
+      this.jobDone.await();
```

Comment:
According to the comments in PipelineResult#waitUntilFinish(Duration duration), we should time out and return the current state after the specified time, rather than waiting for the job to finish.

(The NEXMark benchmark uses this feature to collect metrics periodically)

Reply:
Sounds good. I was unsure that it gets called after the job finishes even without a timeout.